### PR TITLE
Added new extensions to existing syntax highlighting

### LINKED ISSUE #17503: No syntax highlight for .cc files (c++)
### The problem

Should be self-descriptive. Not sure if I'm missing something.
image

### Release version

Version 3.3.4-beta1 (x64)

### Operating system

Windows 11

### PR DESCRIPTION
**Pull request by DylanDevelops**
Closes #17503

## Description

- Added support for the following files:
  - CMake (.cmake)
  - C++ Files (.cxx, .hh, .hxx)
  - XML Schema (.xsd)

Let me know if these are correctly implemented. I didn't add any new modules but used existing ones to update syntax highlighting for these new file types. For C++ files, it was easy or XML, but for files like bat CMD and CMAKE, let me know if I didn't implement them correctly.

## Release notes

Notes: [Improved] Syntax highlighting now supports .cmake, .cxx, .hh, .hxx, and .xsd files.

### CODE REVIEW

File: app/src/highlighter/index.ts [hunk 1]
```diff
@@ -149,6 +150,10 @@ const extensionModes: ReadonlyArray<IModeDefinition> = [
       '.cpp': 'text/x-c++src',
       '.hpp': 'text/x-c++src',
       '.cc': 'text/x-c++src',
+      '.hh': 'text/x-c++src',
+      '.hxx': 'text/x-c++src',
+      '.cxx': 'text/x-c++src',
+      '.cmake': 'text/x-c++src',
```
Comment (niik) [inline app/src/highlighter/index.ts]: https://codemirror.net/5/mode/cmake/index.html defines this as `text/x-cmake`

File: app/src/highlighter/index.ts [hunk 2]
```diff
@@ -172,6 +177,8 @@ const extensionModes: ReadonlyArray<IModeDefinition> = [
     install: () => import('codemirror/mode/shell/shell'),
     mappings: {
       '.sh': 'text/x-sh',
+      '.bat': 'text/x-sh',
+      '.cmd': 'text/x-sh',
```
Comment (niik) [inline app/src/highlighter/index.ts]: `.bat` and `.cmd` are a whole different language from that of shell script and that language is not supported in codemirror at this time, see https://github.com/codemirror/codemirror5/issues/5497

**Review by niik — DISMISSED** [review]
Hey @DylanDevelops, thanks for wanting to contribute. Please see https://github.com/desktop/desktop/blob/development/docs/technical/syntax-highlighting.md#i-want-to-add-my-favorite-language for step by step instructions on adding new languages (or extensions) for syntax highlighting and please open a PR against https://github.com/desktop/highlighter-tests/ with sample files for the extensions you've added.

**Comment by DylanDevelops** [comment]
I'll get on this soon! 👍🏼

File: app/src/highlighter/index.ts (hunk 2 above)
Comment (DylanDevelops) [reply app/src/highlighter/index.ts]: Removed .bat and .cmd from this pull request

**Comment by DylanDevelops** [comment]
@niik Do you know why, after this most recent commit, CMake files aren't getting the syntax highlighting it should?

File: app/src/highlighter/index.ts (hunk 1 above)
Comment (DylanDevelops) [reply app/src/highlighter/index.ts]: Removed `x-cmake` from CLike and created a new install specific for CMake.

**Comment by Salm3** [comment]
@

**Comment by DylanDevelops** [comment]
Any updates on this?

**Comment by niik** [comment]
> @niik Do you know why, after this most recent commit, CMake files aren't getting the syntax highlighting it should?

I'm not sure what you mean by "the syntax highlighting it should" but it appears as if there's syntax highlighting being applied:

image
It appears that the cmake syntax mode in code mirror is very basic: https://github.com/codemirror/codemirror5/blob/master/mode/cmake/cmake.js

It seems that the cmake mode uses the `def` token to describe functions and we've explicitly turned off highlighting for that due to other modes using it too liberally. I pushed an update to your branch that includes a css rule that should make functions colorized. Take a look and let me know what you think.

**Comment by DylanDevelops** [comment]
Ah okay. That makes more sense, and it works for me! I appreciate your help, @niik!

Do you need anything else from me for this pull request?

**Review by sergiou87 — APPROVED** [review]
Works as expected, thank you

**Pull request merged**